src/librustc_errors/diagnostic.rs

```diff
@@ -170,6 +166,28 @@ impl Diagnostic {
         self
     }
 
+    /// Prints out a message with one or multiple suggested edits of the
+    /// code, which may break the code, require manual intervention
+    /// or be plain out wrong, but might possibly be the correct solution.
+    ///
+    /// See `diagnostic::RenderSpan::Guesses` for more information.
+    pub fn guesses<I>(&mut self, sp: Span, msg: &str, guesses: I) -> &mut Self
+        where I: IntoIterator<Item = String>
+    {
+        self.sub(Level::Help,
+                 msg,
+                 MultiSpan::new(),
+                 Some(Guesses(guesses.into_iter().map(|guess| CodeSuggestion {
+                     msp: sp.into(),
+                     substitutes: vec![guess],
+                 }).collect())));
+        self
+    }
+
+    pub fn guess(&mut self, sp: Span, msg: &str, guess: String) -> &mut Self {
```

Comment:
I thinks `span_guess` is a better name, because all other functions taking a span have span in the name. ( for exanple `span_suggestion` )